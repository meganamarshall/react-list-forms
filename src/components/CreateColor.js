import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class CreateColor extends PureComponent {
  static propTypes = {
    addColor: PropTypes.func.isRequired
  }

  state = {
    name: '',
    color: '#000000'
  }

  handleSubmit = event => {
    event.preventDefault();
    const { name, color } = this.state;

    this.props.addColor({ name, hex: color });
  }

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  }

  render() {
    const {
      name,
      color
    } = this.state;

    return (
      <form onSubmit={this.handleSubmit}>
        <input name="name" type="text" value={name} onChange={this.handleChange} />
        <input name="color" type="color" value={color} onChange={this.handleChange} />
        <button>Add Color!</button>
      </form>
    );
  }
}
