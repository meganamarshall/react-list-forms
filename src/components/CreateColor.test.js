import React from 'react';
import { shallow } from 'enzyme';
import CreateColor from './CreateColor';

describe('create color component', () => {
  it('renders the create color', () => {
    const wrapper = shallow(<CreateColor addColor={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('updates name', () => {
    const wrapper = shallow(<CreateColor addColor={jest.fn()} />);

    wrapper.find('[name="name"]').at(0).simulate('change', {
      target: { name: 'name', value: 'pink' }
    });
  });

  it('updates color', () => {
    const wrapper = shallow(<CreateColor addColor={jest.fn()} />);

    wrapper.find('[name="color"]').at(0).simulate('change', {
      target: { name: 'color', value: '54kshgi' }
    });
  });
});

