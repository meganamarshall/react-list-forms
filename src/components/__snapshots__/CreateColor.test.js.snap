// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create color component renders the create color 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="name"
    onChange={[Function]}
    type="text"
    value=""
  />
  <input
    name="color"
    onChange={[Function]}
    type="color"
    value="#000000"
  />
  <button>
    Add Color!
  </button>
</form>
`;
